spec.md

```diff
@@ -119,27 +117,27 @@ On a theoretical front, Val owes greatly to linear types [(Wadler 1990)](https:/
       decimal-floating-point-literal
       hexadecimal-floating-point-literal
 
-    decimal-floating-point-literal ::=
+    decimal-floating-point-literal ::= (token)
       decimal-fractional-constant exponent?
       decimal-literal exponent
 
-    decimal-fractional-constant ::=
+    decimal-fractional-constant ::= (token)
       decimal-literal . decimal-literal
 
-    exponent ::=
-      e exponent-sign? decimal-literal
-      E exponent-sign? decimal-literal
+    exponent := (token)
+      'e' exponent-sign? decimal-literal
+      'E' exponent-sign? decimal-literal
 
-    hexadecimal-floating-point-literal ::=
-      hexadecimal-fractional-constant binary-exponent
+    hexadecimal-floating-point-literal ::= (token)
+      hexadecimal-fractional-constant binary-exponent?
```

Comment:
swiftc tells me the exponent is necessary when writing a hexadecimal floating-point literal. Is it wrong?

Reply:
🤷 it's your grammar, so it can say whatever you want.  I don't know whether it _should_ be necessary, but the asymmetry in your original grammar caught me by surprise.  I asked about it in an overleaf comment I think you haven't replied to those in quite a while.

Reply:
I have to admit I don't know how hexadecimal floating point literal work. I never used them and never seen them, even in hacky 3D graphics/physics engines. I tried to build an understanding from observing Swift and C++. I wouldn't mind if we remove them altogether.

Reply:
That's up to you, braddah.  I'm just trying to get the grammar straightened out.

Reply:
Out they go!